tests/unit/splinterdb_quick_test.c

```diff
@@ -437,6 +437,8 @@ CTEST2(splinterdb_quick, test_basic_iterator)
       ASSERT_EQUAL(0, rc);
       i++;
    }
+   ASSERT_EQUAL(num_inserts, i);
```

Comment:
Really ... unrelated to this fix. But while in this file, decided to augment some checks verifying that we are returning right # of kv-pairs.